doc/compiler/il/notes_mainline_outOfLine_GlRegDeps.md

```diff
@@ -0,0 +1,97 @@
+```
+n541n     treetop                                                                             [0x7fec3b0b98d0] bci=[1,29,-] rc=0 vc=0 vn=- li=- udi=- nc=1
+n190n       iand (X>=0 cannotOverflow )                                                       [0x7fec3b0b2b20] bci=[1,29,-] rc=5 vc=1095 vn=- li=- udi=- nc=2 flg=0x1100
+n185n         iadd (X>=0 cannotOverflow )                                                     [0x7fec3b0b2990] bci=[1,21,-] rc=1 vc=1095 vn=- li=- udi=- nc=2 flg=0x1100
+n183n           iload  n<auto slot 5>[#370  Auto] [flags 0x3 0x0 ] (X>=0 cannotOverflow )     [0x7fec3b0b28f0] bci=[1,18,-] rc=1 vc=1095 vn=- li=- udi=34 nc=0 flg=0x1100
+n184n           iconst -1 (X!=0 X<=0 )                                                        [0x7fec3b0b2940] bci=[1,20,-] rc=1 vc=1095 vn=- li=- udi=- nc=0 flg=0x204
+n505n         ==>iRegLoad
+n542n     treetop                                                                             [0x7fec3b0b9920] bci=[1,17,-] rc=0 vc=0 vn=- li=- udi=- nc=1
+n506n       ==>aRegLoad
+n558n     iRegStore edx                                                                       [0x7fec3b0b9e20] bci=[1,29,-] rc=0 vc=0 vn=- li=- udi=- nc=1
+n190n       ==>iand
+n582n     ificmpne --> block_57 BBStart at n553n ()                                           [0x7fec3b0ba5a0] bci=[1,30,-] rc=0 vc=0 vn=- li=- udi=- nc=3 flg=0x20
+n580n       iand                                                                              [0x7fec3b0ba500] bci=[1,30,-] rc=1 vc=0 vn=- li=- udi=- nc=2
+n578n         iloadi  <isClassFlags>[#287  Shadow +36] [flags 0x603 0x0 ]                     [0x7fec3b0ba460] bci=[1,30,-] rc=1 vc=0 vn=- li=- udi=- nc=1
+n577n           aloadi  <componentClass>[#282  Shadow +88] [flags 0x10607 0x0 ]               [0x7fec3b0ba410] bci=[1,30,-] rc=1 vc=0 vn=- li=- udi=- nc=1
+n576n             aloadi  <vft-symbol>[#288  Shadow] [flags 0x18607 0x0 ]                     [0x7fec3b0ba3c0] bci=[1,30,-] rc=1 vc=0 vn=- li=- udi=- nc=1
+n506n               ==>aRegLoad
+n579n         iconst 1024                                                                     [0x7fec3b0ba4b0] bci=[1,30,-] rc=1 vc=0 vn=- li=- udi=- nc=0
+n581n       iconst 0                                                                          [0x7fec3b0ba550] bci=[1,30,-] rc=1 vc=0 vn=- li=- udi=- nc=0
+n583n       GlRegDeps ()                                                                      [0x7fec3b0ba5f0] bci=[1,30,-] rc=1 vc=0 vn=- li=- udi=- nc=3 flg=0x20
+n506n         ==>aRegLoad
+n505n         ==>iRegLoad
+n584n         PassThrough rdx                                                                 [0x7fec3b0ba640] bci=[1,29,-] rc=1 vc=0 vn=- li=- udi=- nc=1
+n190n           ==>iand
+n572n     BBEnd </block_33>                                                                   [0x7fec3b0ba280] bci=[1,30,-] rc=0 vc=0 vn=- li=- udi=- nc=0
+```
+
```

Comment:
It is a little odd that the wrong and right examples above are not using the same basic set of IL trees. If I remember right you had a bug in the past couple of weeks where you had an evaluation of an arithmetic node (e.g. iadd) under the `GlRegDeps` and you fixed that later. This could be the example you use with the same basic set of IL trees showing the wrong and right way of doing the same thing.

Reply:
I'll match the good example and the bad from the same method compilation in the next update.